src/fondant/component_spec.py

```diff
@@ -18,52 +17,50 @@
 from fondant.exceptions import InvalidComponentSpec
 from fondant.schema import Field, KubeflowCommandArguments, Type
 
-# # TODO: remove after upgrading to kfpv2
-
-kubeflow_to_python_type_dict = {
-    "STRING": str,
-    "NUMBER_INTEGER": int,
-    "NUMBER_DOUBLE": float,
-    "BOOLEAN": lambda x: bool(strtobool(x)),
-    "STRUCT": json.loads,
-    "LIST": json.loads,
-}
-
-
-def kubeflow2python_type(type_: str) -> t.Any:
-    map_fn = kubeflow_to_python_type_dict[type_]
-    return lambda value: map_fn(value) if value != "None" else None  # type: ignore
-
-
-python2kubeflow_type = {
-    "str": "STRING",
-    "int": "NUMBER_INTEGER",
-    "float": "NUMBER_DOUBLE",
-    "bool": "BOOLEAN",
-    "dict": "STRUCT",
-    "list": "LIST",
-}
-
 
 @dataclass
 class Argument:
```

Comment:
I feel like we could do more with this class:
- handle argument conversion to argparse and kfp
- do type checking (sane defaults)
- handle default/optional logics